### PR TITLE
Add reusable actions for Dash license checking

### PR DESCRIPTION
**Pull request by pstuecker**
This PR provides two new reusable workflows

- dash.yml Provides license verification for PRs
- update-dependencies.yml updates the `DEPENDENCIES` file. This is to be used for development branches to update license information during development. 

Also the existing `set-version` action is updated to also update the `DEPENDENCIES` file whenever a release branch/tag is created.

### CODE REVIEW

File: .github/workflows/release-branch.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
     - name: Bump versions
       uses: eclipse-set/build/.github/actions/set-version@main
       with:
-        version: ${{ inputs.version }}.0-SNAPSHOT
+        version: ${{ inputs.version }}.1-SNAPSHOT
```
Comment (PetersNils) [inline .github/workflows/release-branch.yml:34]: What's the reason behind this change?

File: .github/workflows/update-dependencies.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+jobs:        
+  release-tag:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write  
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Bump versions
+      uses: eclipse-set/build/.github/actions/update-dependency-file@main
```
Comment (PetersNils) [inline .github/workflows/update-dependencies.yml:21]: I guess this is an c&p error?
```suggestion
    - name: Update DEPENDENCIES
      uses: eclipse-set/build/.github/actions/update-dependency-file@main
```

File: .github/workflows/dash.yml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      uses: eclipse-set/build/.github/actions/dash-license-check-action@main
+      with:
+        request-review: ${{ env.request-review }}
+        project-id: ${{ inputs.projectId }}
+        pom: ${{ inputs.pom }}
+      env:
+        GITLAB_API_TOKEN: ${{ secrets.gitlabAPIToken }}
+    
+    # Hide outdated License comments
+    - uses: int128/hide-comment-action@5aa4ad6cf762098ebdc5d3c046e9e1fb35fe5ce8 # v1
```
Comment (PetersNils) [inline .github/workflows/dash.yml:105]: Comment full version tag.
```suggestion
    - uses: int128/hide-comment-action@5aa4ad6cf762098ebdc5d3c046e9e1fb35fe5ce8 # v1.23.0
```
Btw: 1.24.0 has been released today

**Review by PetersNils — CHANGES_REQUESTED** [review]

File: .github/workflows/dash.yml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    
+    - name: Check license vetting status (and ask for review if requested)
+      id: check-license-vetting
+      uses: eclipse-set/build/.github/actions/dash-license-check-action@main
+      with:
+        request-review: ${{ env.request-review }}
+        project-id: ${{ inputs.projectId }}
+        pom: ${{ inputs.pom }}
+      env:
+        GITLAB_API_TOKEN: ${{ secrets.gitlabAPIToken }}
```
Comment (PetersNils) [inline .github/workflows/dash.yml:102]: I guess this requires a PR within eclipse-set/.eclipsefdn, correct?

Comment (pstuecker) [reply .github/workflows/dash.yml:102]: I'm not entirely sure if this wasn't already set up during our gitlab -> github migration. This has just been taken from the original dash action

File: .github/workflows/release-branch.yml (hunk 1 above)
Comment (pstuecker) [reply .github/workflows/release-branch.yml:34]: I mixed up at which point we need to update the release branch development version. Reverted.

File: .github/workflows/dash.yml (hunk 3 above)
Comment (pstuecker) [reply .github/workflows/dash.yml]: Updated to 1.24.0

File: .github/workflows/dash.yml (hunk 4 above)
Comment (PetersNils) [reply .github/workflows/dash.yml:102]: We can give this a try, but I'm expecting this to not work.

**Review by PetersNils — APPROVED** [review]
